### PR TITLE
Format installer and archive file name per new guidelines

### PR DESCRIPTION
**Pull request by rakeshsinghranchi**
Guidance for naming convention : https://github.com/dotnet/designs/issues/2 

This commit takes care of the following : 
1. Shifting rid to the end
2. Adding runtime to the bundle packages and archives

**Pending**:
1. Updating the rid , for e.g. for Windows : win10-{arch} , osx : macos.10.12-x64 . 
2. Renaming sharedframework artifacts to "runtime" . This needs further discussion as it will collide with bundle installer name.

This needs further discussion as portable assets as of today do not carry OS version number.

With this change :
for e.g.
Bundle:
dotnet-runtime-2.0.0-preview2-25319-9-win-x86.exe

MSIs:
dotnet-host-2.0.0-preview2-23519-9-win-x86.msi
dotnet-hostfxr-2.0.0-preview2-23519-9-win-x86.msi
dotnet-sharedframework-2.0.0-preview2-23519-9-win-x86.msi

Archive:
dotnet-host-2.0.0-preview2-23519-9-win-x86.zip
dotnet-hostfxr-2.0.0-preview2-23519-9-win-x86.zip
dotnet-sharedframework-2.0.0-preview2-23519-9-win-x86.zip
dotnet-runtime-2.0.0-preview2-25319-9-win-x86.zip

Note: Ask for name format update is for installers and archives only .

### CODE REVIEW

**Comment by rakeshsinghranchi** [comment]
@gkhanna79  , there are pending items to close on before sending this issue to shiproom:

1. Update the installer file name to show OS version. As of today, OS name is derived from TargetRid and TargetRid for portable assets do not carry OS version.  
If we agree to add the OS version number, do you think updating this is enough or it will have more involved changes ?
https://github.com/dotnet/core-setup/blob/release/2.0.0/dir.props#L226

2. Guidance document proposes to rename sharedframework installer to runtime. for e.g.
Current : dotnet-sharedframework-ubuntu.16.04-x64.2.0.0-preview2-25319-01.deb 
Proposed:dotnet-**runtime**-2.0.0-preview2-25319-01-ubuntu.16.04-x64.deb

This will cause collusion for Windows and OSX where we build bundle packages with same name.

**Comment by Petermarcu** [comment]
@rakeshsinghranchi @leecow @richlander Lets get in a room on Monday and make sure we're all on the same page and all questions are answered.

**Comment by rakeshsinghranchi** [comment]
@dotnet-bot  test Ubuntu16.04 x64 Release Build please

File: dir.props [hunk 1]
```diff
@@ -287,9 +287,17 @@
   </PropertyGroup>
   
   <PropertyGroup>
+    <CombinedInstallerStart>$(PackagesOutDir)dotnet-runtime-</CombinedInstallerStart>
     <SharedHostInstallerStart>$(PackagesOutDir)dotnet-host-</SharedHostInstallerStart>
     <HostFxrInstallerStart>$(PackagesOutDir)dotnet-hostfxr-</HostFxrInstallerStart>
-    <SharedFrameworkInstallerStart>$(PackagesOutDir)dotnet-sharedframework-</SharedFrameworkInstallerStart>
+    <SharedFrameworkInstallerStart>$(PackagesOutDir)dotnet-runtime-</SharedFrameworkInstallerStart>
+
+    <!-- OSX specific intermediate package suffix -->
```
Comment (eerhardt) [inline dir.props]: I'm not sure I understand what this is for/why it is needed.  Can this comment explain why this is doing what it is doing?

**Comment by rakeshsinghranchi** [comment]
cc @leecow 
Sent email to shiproom for approval. Waiting for approval. Note that there will be a follow up PR to update readme file in master branch.

**Pull request merged**